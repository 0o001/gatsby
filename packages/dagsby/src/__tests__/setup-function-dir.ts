const setupFunctionDir = require(`../setup-function-dir`)
const os = require(`os`)
const uuid = require(`uuid`)
const path = require(`path`)
const fs = require(`fs-extra`)

describe(`setup-function-dir`, () => {
  it(`sets up a function dir`, async () => {
    const aFunction = args => args.a
    const id = uuid.v4()
    const directory = path.join(os.tmpdir(), id)
    fs.ensureDirSync(directory)
    const funcPath = path.join(directory, `index.js`)
    setupFunctionDir({
      task: {
        func: aFunction.toString(),
        digest: 1,
      },
      funcDir: directory,
      funcPath,
    })

    const funcStr = await fs.readFile(funcPath, `utf-8`)

    expect(funcStr).toEqual(`module.exports = ${aFunction.toString()}`)
  })

  it(`installs dependencies for a task`, async () => {
    const aFunction = args => args.a
    const id = uuid.v4()
    const directory = path.join(os.tmpdir(), id)
    fs.ensureDirSync(directory)
    const funcPath = path.join(directory, `index.js`)
    await setupFunctionDir({
      task: {
        func: aFunction.toString(),
        digest: 1,
        dependencies: {
          lodash: `latest`,
        },
      },
      funcDir: directory,
      funcPath,
    })

    const funcStr = await fs.readFile(funcPath, `utf-8`)
    const packageJson = await fs.readFile(
      path.join(directory, `package.json`),
      `utf-8`
    )
    const parsedJson = JSON.parse(packageJson)

    expect(funcStr).toEqual(`module.exports = ${aFunction.toString()}`)
    expect(parsedJson).toMatchSnapshot()
    expect(parsedJson.dependencies).toHaveProperty(`lodash`)
  })
})
