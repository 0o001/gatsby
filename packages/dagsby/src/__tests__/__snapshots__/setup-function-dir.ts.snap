// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setup-function-dir installs dependencies for a task 1`] = `
Object {
  "author": "",
  "dependencies": Object {
    "lodash": "^4.17.20",
  },
  "description": "",
  "keywords": Array [],
  "license": "ISC",
  "main": "index.js",
  "name": "e93a3f80-e996-4d2c-b308-b4fb6d6e19f6",
  "scripts": Object {
    "test": "echo \\"Error: no test specified\\" && exit 1",
  },
  "version": "1.0.0",
}
`;
